detox/src/devices/android/ADB.js

```diff
@@ -283,4 +287,17 @@ class ADB {
   }
 }
 
+function _reportTelnetPortResolutionError(input, devicesList, deviceAdbName, port) {
+  const log = require('../../utils/logger').child({ __filename });
+  const {encodeBase64} = require('../../utils/encoding');
+  log.error({event: 'DEVICE_NAME_ERROR'}, `Failed to determine port for emulator device '${deviceAdbName}!!!`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `Please help us out by reporting all DEVICE_NAME_ERROR logs in: https://github.com/wix/Detox/issues/1427`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `State dump ==>`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `adb devices: ${input}`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `adb devices (base64): '${encodeBase64(input)}'`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `devicesList: ${devicesList}`);
+  log.error({event: 'DEVICE_NAME_ERROR'}, `port: ${port}`);
+  throw new Error(`Unable to determine port for emulator device '${deviceAdbName}'!`);
```

Comment:
I suggest extracting `throw new Error` out of `_report...` method, to the `if` branch above.